plugins/dell-dock2/fu-dell-dock2-ec.c

```diff
@@ -0,0 +1,972 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-dell-dock2-common.h"
+
+#define BIT_SET(x, y)	(x |= (1 << y))
+#define BIT_CLEAR(x, y) (x &= (~(1 << y)))
+
+const FuHIDI2CParameters dell_dock2_ec_base_settings = {
+    .i2ctargetaddr = DELL_DOCK2_EC_HID_I2C_ADDRESS,
+    .regaddrlen = 1,
+    .i2cspeed = I2C_SPEED_250K | 0x80,
```

Comment:
I notice there are definitions for 400k and 800k.  Have you already tried these?  Have you talked to internal team about upgrading to them?  If any of them are more stable, it would be ideal to use something faster.

Reply:
My apologies. For K2, the I2C speed is actually unused, the communication to EC will be done directly via ubshid, so we don't need to route through the rtshub endpoint. I will remove the settings but keep the structure in the header. Sorry for the confusion.

Reply:
Ah I thought it was from another internal bus, but that makes sense, thanks for clarifying.

Reply:
It has come to my attention that the addr settings is useful for a better backward support for the predecessors, i.e. we can read the dock type, to determine support or no from this plugin.  I will send another PR to bring it back.